tmt/steps/finish/__init__.py

```diff
@@ -22,8 +21,9 @@ class Finish(tmt.steps.Step):
     steps failed (for example when the environment preparation was not
     successful) so that provisioned systems are not kept running.
     """
+    data: List[StepData]
```

Comment:
@guy9050 why is this needed, data is defined in `tmt.steps.Step` and it has `Union[StepData, List[StepData]]` type?

Reply:
It's something I try to address in #1384, `self.data` may be either a single object or a list of objects - that's what annotations say, but in reality, it's always a list -, and often a particular plugin treats `self.data` as the list only. A developer focusing on that particular plugin cannot resolve the original issue, and adding `List[StepData]` makes annotating the plugin easier.

@janhavlin mentioned these in #1384, and I'll resolve them in my patch (once I'm done with failing tests...).

Reply:
@thrix does that answer your question?

Reply:
yep, ty!